docs/JLBP-0016.md

```diff
@@ -11,58 +11,56 @@ Upper version alignment ensures that upgrading packages to compatible higher ver
 
 ## Achieving upper version alignment
 
-Upper version alignment increases the likelihood that consumers' build systems
+Upper version alignment increases the likelihood that build systems
 select the right versions of direct and transitive dependencies, reducing the
 number of conflicts.
 
 See the details specific to each build system in the following sections.
 
 ## Maven
 
-Use `requireUpperBoundDeps` enforcement to ensure that you are using the
+Use the [`requireUpperBoundDeps` Maven enforcer
+rule](https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html)
+to ensure that you are using the
 highest version of each dependency in your dependency tree.
 
+For any transitive dependency that fails the `requireUpperBoundDeps` check, add
+the dependency as a direct dependency so that the path to the correct version
+is shorter, leading Maven to select it instead of the wrong version.
+
 To ensure that dependencies between modules in the project are consistent,
-the parent POM should import the library's own BOM into its
-`<dependencyManagement>` section.
+the project [should publish a BOM](JLBP-0015.md) and
+the parent POM should import this BOM in its `<dependencyManagement>` section.
+Each module's POM should inherit from the parent POM of the library.
+
 
   - [Example import in google-cloud-java](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/36409f5b1df89609eaef92d09cebea97931339bd/google-cloud-clients/pom.xml#L174).
 
-To ensure that usage of dependencies is consistent, for any direct
-dependencies that have a BOM of their own, import those BOMs in a
-`<dependencyManagement>` section.
+To ensure consistency of dependencies from outside the project:
 
-  - If multiple imported BOMs manage the same dependency, use the
+  - If the dependency publishes a BOM, import that BOM in the `<dependencyManagement>` section.
+  
+    If multiple imported BOMs manage the same dependency, use the
     `<dependencyManagement>` of the parent POM to select a compatible version,
     typically the highest.
-  - Order doesn't matter for the override, because all explicit version
+    Order doesn't matter for the override, because all explicit version
     declarations in `<dependencyManagement>` take precedence over all BOM
-    imports. (Order between BOMs matters - the earliest import of a dependency
-    takes precedence over later imports of the same dependency.)
+    imports. Order between BOMs does matter. The first import of a dependency
+    takes precedence over later imports of the same dependency.
 
-For direct dependencies that don't have a BOM (for example, Joda-Time), make
-sure only one pom.xml defines the version, so that the library doesn't
-accidentally depend on different versions in different modules.
+  - If a dependency does not have have a BOM (for example, Joda-Time), make
```

Comment:
Nice. Making this as a bullet point highlights the two cases of "with BOM" and "without BOM".